src/plugins/lua/spamassassin.lua

```diff
@@ -1443,28 +1436,28 @@ local function post_process()
         end
       end
     end,
-    _.filter(function(k, r)
+    filter(function(k, r)
         return r['type'] == 'meta'
       end,
       rules))
 
   -- Check meta rules for foreign symbols and register dependencies
-  _.each(function(k, r)
+  each(function(k, r)
       if r['expression'] then
         local expr_atoms = r['expression']:atoms()
 
         for i,a in ipairs(expr_atoms) do
-          if not atoms[a] then
```

Comment:
It is slightly wrong. You need to register dependency not on `a` but on external dependencies of `a`. Furthermore, you need to adjust the current external deps for this meta to include dependencies from `a`. And finally, you need to perform this operation unless there are no changes in deps lists.